kevm-pyk/src/kevm_pyk/kproj/evm-semantics/lemmas/evm-int-simplification.k

```diff
@@ -20,36 +20,36 @@ module EVM-INT-SIMPLIFICATION-COMMON
     rule [upInt-lt-true]:
       ((X up/Int Y) *Int Y) <Int Z => true
       requires X +Int Y <=Int Z andBool 0 <Int Y
-      [simplification]
+      [simplification, preserves-definedness]
```

Comment:
Agree the definedness could be inferred from side conditions for many of the equations that I marked manually here.
Unfortunately we had to back out @goodlyrottenapple s change to do this inference when loading the definition, because it caused problems (which we are in the process of fixing). We hope to get it back soon but I wanted to make progress on this in the meanwhile.